File: frontend/src/types/task.ts
// 基础任务类型
export interface Task {
  id: number
  name: string
  type: string
  phase: string
  status: string
  description: string
  create_time: string
  update_time: string
}

// 任务表单类型
export interface TaskForm {
  name: string
  type: string  
  phase: string
  description: string
}

// 任务分配类型
export interface TaskAssignment {
  id: number
  task_id: number
  user_id?: number
  username: string
  status: string
  progress: number
  performance_score: number
  comments?: string
  assigned_at: string
  last_update: string
  // 任务信息
  task_name?: string
  task_type?: string
  task_phase?: string
}

// 任务分配表单类型
export interface TaskAssignmentForm {
  task_id: number
  username: string
  comments?: string
}

// 任务进度类型
export interface TaskProgress {
  id: number
  task_id: number
  task_name: string
  task_type: string
  username: string
  status: string
  progress: number
  performance_score: number
  comments: string
  assigned_at: string
  last_update: string
}

// 任务进度更新类型
export interface TaskProgressUpdate {
  status?: string
  progress?: number
  performance_score?: number
  comments?: string
}

// 任务统计类型
export interface TaskStats {
  total: number
  inProgress: number
  completed: number
  paused: number
  avgProgress: number
}

// 图表数据类型
export interface TaskChartData {
  statusDistribution: {
    name: string
    value: number
  }[]
  progressTrend: {
    name: string
    value: number
  }[]
}

// 用户类型（用于任务分配）
export interface TaskUser {
  id: number
  username: string
  role: string
  type: string
}

// 可用任务类型（用于任务分配）
export interface AvailableTask {
  id: number
  name: string
  type: string
  phase: string
  description: string
}

// 任务历史记录类型
export interface TaskHistory {
  id: number
  assignment_id: number
  status: string
  progress: number
  performance_score: number
  comments: string
  updated_at: string
  updated_by: string
}

// 任务状态枚举
export enum TaskStatus {
  UNASSIGNED = '未分配',
  IN_PROGRESS = '进行中',
  COMPLETED = '已完成',
  PAUSED = '已暂停'
}

// 任务类型枚举
export enum TaskType {
  NETWORK_BUILD = '网络搭建任务',
  SYSTEM_BUILD = '系统构建任务', 
  OPERATION_MONITOR = '运维监管任务',
  LOG_SECURITY = '日志安全任务'
}

// 任务阶段枚举
export enum TaskPhase {
  PLANNING = '计划阶段',
  EXECUTION = '执行阶段',
  ACCEPTANCE = '验收阶段',
  COMPLETION = '完成阶段'
}

// API响应类型
export interface TaskListResponse {
  items: Task[]
  total: number
  page: number
  size: number
}

export interface TaskAssignmentListResponse {
  items: TaskAssignment[]
  total: number
  page: number
  size: number
}

export interface TaskProgressListResponse {
  items: TaskProgress[]
  total: number
  page: number
  size: number
}

// 搜索参数类型
export interface TaskSearchParams {
  name?: string
  type?: string
  phase?: string
  page?: number
  size?: number
}

export interface TaskAssignmentSearchParams {
  taskName?: string
  username?: string
  status?: string
  page?: number
  size?: number
}

export interface TaskProgressSearchParams {
  taskName?: string
  username?: string
  status?: string
  dateRange?: string[]
  page?: number
  size?: number
} 

File: backend/app/schemas/task.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class TaskBase(BaseModel):
    """任务基础模式"""
    name: str
    type: Optional[str] = None
    phase: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = '未分配'

class TaskCreate(TaskBase):
    """创建任务模式"""
    pass

class TaskUpdate(BaseModel):
    """更新任务模式"""
    name: Optional[str] = None
    type: Optional[str] = None
    phase: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None

class Task(TaskBase):
    """任务响应模式"""
    id: int
    create_time: datetime
    update_time: datetime
    
    model_config = ConfigDict(from_attributes=True)

class TaskInDB(Task):
    """数据库中的任务模式"""
    pass

# 批量导入结果
class TaskBulkImportResult(BaseModel):
    """任务批量导入结果"""
    success_count: int
    fail_count: int
    failed_tasks: List[dict]
    message: str

class TaskAssignmentBase(BaseModel):
    """任务分配基础模式"""
    task_id: int
    user_id: Optional[int] = None
    username: str
    status: Optional[str] = '进行中'
    progress: Optional[int] = 0
    performance_score: Optional[int] = 0
    comments: Optional[str] = None

class TaskAssignmentCreate(TaskAssignmentBase):
    """任务分配创建模式"""
    pass

class TaskAssignmentUpdate(BaseModel):
    """任务分配更新模式"""
    status: Optional[str] = None
    progress: Optional[int] = None
    performance_score: Optional[int] = None
    comments: Optional[str] = None

class TaskAssignmentResponse(TaskAssignmentBase):
    """任务分配响应模式"""
    id: int
    assigned_at: datetime
    last_update: datetime
    # 任务信息
    task_name: Optional[str] = None
    task_type: Optional[str] = None
    task_phase: Optional[str] = None
    
    model_config = ConfigDict(from_attributes=True)

class TaskWithAssignments(Task):
    """包含分配信息的任务模式"""
    assignments: List[TaskAssignmentResponse] = [] 